go-controller/pkg/clustermanager/network_cluster_controller.go

```diff
@@ -135,7 +159,40 @@ func (ncc *networkClusterController) init() error {
 	if ncc.hasPodAllocation() {
 		ncc.retryPods = ncc.newRetryFramework(factory.PodType, true)
 
-		ncc.podAllocator = pod.NewPodAllocator(ncc.NetInfo, ncc.watchFactory.PodCoreInformer().Lister(), ncc.kube)
+		if ncc.allowPersistentIPs() {
+			ncc.retryIPAMClaims = ncc.newRetryFramework(factory.IPAMClaimsType, true)
+		}
+
+		var (
+			podAllocationAnnotator annotationalloc.AnnotationAllocator = annotationalloc.NewPodAnnotationAllocator(
+				ncc.NetInfo,
+				ncc.watchFactory.PodCoreInformer().Lister(),
+				ncc.kube,
+			)
+			ipamClaimsFetcher persistentips.Fetcher
+		)
+
+		if ncc.allowPersistentIPs() {
+			ipamClaimsAllocator := persistentips.NewIPAMClaimReconciler(ncc.kube, ncc.NetInfo, ncc.watchFactory.IPAMClaimsInformer().Lister())
```

Comment:
nit: `ipamClaimsReconciler` instead of `ipamClaimsAllocator` (also in the other instaces for the non-interconnect side of things)